src/jenesis/contracts/deploy.py

```diff
@@ -153,13 +197,25 @@ def deploy_contracts(cfg: Config, project_path: str, deployer_key: Optional[str]
         run_local_node(selected_profile.network)
 
     contracts = detect_contracts(project_path)
+    contracts_list = {contract.name: contract for contract in contracts}
 
-    contracts_to_deploy = []  # type: List[Tuple[Contract, Deployment]]
     profile_contracts = selected_profile.contracts
     profile_contract_names = list(profile_contracts.keys())
 
-    # determine what tasks to do
-    for contract in contracts:
+    init_addresses = {
+        name: set(profile_contracts[name]["init_addresses"])
+        for name in contracts_list.keys()
+    }
+
+    sorter = gl.TopologicalSorter(init_addresses)
+    deployment_order = list(sorter.static_order())
+    contract_to_deploy = ""
```

Comment:
It's not clear to me why `contract_to_deploy` is necessary in the loop below. I also would've expected it to get reset to `""` at the beginning of each loop iteration.

Would it be a bit more straightforward to simply `continue` the loop whenever one of the conditions (configuration not out of date or digest not changed) holds?

Reply:
Yeah that is a good idea! Ill get to it